src/library/scala/collection/Factory.scala

```diff
@@ -408,6 +420,27 @@ trait EvidenceIterableFactory[+CC[_], Ev[_]] extends Serializable {
     */
   def tabulate[A : Ev](n: Int)(f: Int => A): CC[A] = from(new View.Tabulate(n)(f))
 
+  /** Produces a $coll containing repeated applications of a function to a start value.
+    *
+    *  @param start the start value of the $coll
+    *  @param len   the number of elements contained in the $coll
+    *  @param f     the function that's repeatedly applied
+    *  @return      a $coll with `len` values in the sequence `start, f(start), f(f(start)), ...`
+    */
+  def iterate[A : Ev](start: A, len: Int)(f: A => A): CC[A] = from(new View.Iterate(start, len)(f))
+
+  /** Produces a $coll that uses a function `f` to produce elements of type `A`
+    * and update an internal state of type `S`.
+    *
+    * @param init State initial value
+    * @param f    Computes the next element (or returns `None` to signal
+    *             the end of the collection)
+    * @tparam A   Type of the elements
+    * @tparam S   Type of the internal state
+    * @return a $coll that produces elements using `f` until `f` returns `None`
+    */
+  def unfold[A : Ev, S](init: S)(f: S => Option[(A, S)]): CC[A] = from(new View.Unfold(init)(f))
```

Comment:
If we add it to `IterableFactory` and `EvidenceIterableFactory` we should also add it to `SortedIterableFactory`, `MapFactory` and `SortedMapFactory`.

Or we can remove it from `XxxFactory` types and let users write `Iterator.unfold(…)(…).to(Xxx)` instead.

Reply:
OK, I’m fine with this argument.

Reply:
@julienrf IMHO it would be best not to expect users to rely on iterators for common useful operations, as iterators should be regarded as an error-prone (stateful) lowish-level abstraction, mainly used as implementation details.

Reply:
@LPTK Then you can just write: `View.unfold(…)(…).to(Xxx)` :)

Reply:
@julienrf you can't, because `object View` inherits `unfold` from `IterableFactory`. You could do `new View.Unfold(...)(...).to(Xxx)`, but that seems not great.